src/renderer/components/AgentBase.vue

```diff
@@ -151,7 +101,21 @@ import { mapState, mapActions } from "vuex";
 import { from_store } from '../connection_detail.js';
 import message_bus from '@/message_bus.js';
 import share, {share_source} from '@/share.js';
-import {shared} from './components.js';
+import components, {shared} from './components.js';
+
+// icons from https://element.eleme.io/#/en-US/component/icon
+let module_list = Object.entries(components).map(([modulename, module]) => ({
+  path: module.default.name,
+  label: module.metadata && module.metadata.menu && module.metadata.menu.label ? module.metadata.menu.label : module.default.name,
```

Comment:
This is a little complicated to parse; I'd opt to make this logic a little more verbose

Reply:
I'll modify a bit and add a comment to explain. Javascript lacks an elvis operator, and this general approach is the workaround for possibly non-existant deep access. Making this more verbose is a 4x line increase and will likely not fix readability.